collectors/statsd.plugin/statsd.c

```diff
@@ -413,52 +367,45 @@ static struct statsd {
 // --------------------------------------------------------------------------------------------------------------------
 // statsd index management - add/find metrics
 
-static int statsd_metric_compare(void* a, void* b) {
-    if(((STATSD_METRIC *)a)->hash < ((STATSD_METRIC *)b)->hash) return -1;
-    else if(((STATSD_METRIC *)a)->hash > ((STATSD_METRIC *)b)->hash) return 1;
-    else return strcmp(((STATSD_METRIC *)a)->name, ((STATSD_METRIC *)b)->name);
-}
-
-static inline STATSD_METRIC *statsd_metric_index_find(STATSD_INDEX *index, const char *name, uint32_t hash) {
-    STATSD_METRIC tmp;
-    tmp.name = name;
-    tmp.hash = (hash)?hash:simple_hash(tmp.name);
+static void dictionary_extensions_delete_callback(const char *name, void *value, void *data) {
+    (void)name;
+    (void)data;
+    // STATSD_INDEX *index = (STATSD_INDEX *)data;
+    STATSD_METRIC *m = (STATSD_METRIC *)value;
+    if(m->type == STATSD_METRIC_TYPE_HISTOGRAM || m->type == STATSD_METRIC_TYPE_TIMER) {
+        freez(m->histogram.ext);
+        m->histogram.ext = NULL;
+    }
 
-    return (STATSD_METRIC *)STATSD_AVL_SEARCH(&index->index, (avl_t *)&tmp);
+    if(m->units) freez(m->units);
```

Comment:
Inside `freez` function we already have a [test](https://github.com/netdata/netdata/blob/3071aa055ccf9629278bd79aa2005291f5ad01c4/libnetdata/libnetdata.c#L142) for the pointers, so I think these `ifs` are not necessary.  Is there a special reason to have these tests?